packages/fast-tooling-react/src/css-editor/border-radius/border-radius.spec.tsx

```diff
@@ -0,0 +1,349 @@
+    });
+    test("should use the `data` prop as the input value if the `borderRadius` is provided", () => {
+        const borderRadiusValue: string = "12px 12px 12px 12px";
+        const rendered: any = mount(
+            <CSSBorderRadius
+                data={{ borderRadius: borderRadiusValue }}
+                managedClasses={managedClasses}
+                onChange={jest.fn()}
+            />
+        );
+
+        expect(
+            rendered.find(`.${managedClasses.cssBorderRadius_input}`).prop("value")
+        ).toBe(borderRadiusValue);
+    });
+    test("should fire the `onChange` event if the input value is updated", () => {
+        const borderRadiusValue: string = "12px 12px 12px 12px";
+        const newBorderRadiusValue: string = "10px";
+        const callback: any = jest.fn();
+        const rendered: any = mount(
+            <CSSBorderRadius
+                data={{ borderRadius: borderRadiusValue }}
+                managedClasses={managedClasses}
+                onChange={callback}
+            />
+        );
+
+        expect(callback).toHaveBeenCalledTimes(0);
+
+        rendered
+            .find(`.${managedClasses.cssBorderRadius_input}`)
+            .simulate("change", { target: { value: newBorderRadiusValue } });
+
+        expect(callback).toHaveBeenCalledTimes(1);
+        expect(callback.mock.calls[0][0]).toEqual({ borderRadius: newBorderRadiusValue });
+    });
+    test("should not change the input from controlled to uncontrolled", () => {
+        const borderRadiusValue: string = "12px 12px 12px 12px";
+        const newBorderRadiusValue: string = void 0;
+        const callback: any = jest.fn();
+        const rendered: any = mount(
+            <CSSBorderRadius
+                data={{ borderRadius: borderRadiusValue }}
+                managedClasses={managedClasses}
+                onChange={callback}
+            />
+        );
+
+        expect(callback).toHaveBeenCalledTimes(0);
+
+        rendered
+            .find(`.${managedClasses.cssBorderRadius_input}`)
+            .simulate("change", { target: { value: newBorderRadiusValue } });
+
+        expect(callback).toHaveBeenCalledTimes(1);
+        expect(callback.mock.calls[0][0]).toEqual({ borderRadius: newBorderRadiusValue });
+
+        rendered.setProps({
+            data: {},
+        });
+
+        expect(
+            rendered.find(`.${managedClasses.cssBorderRadius_input}`).prop("value")
+        ).toBe("");
+    });
+
+    test("should parse the `data` prop and assign individual values when 1 value is passed", () => {
+        const borderRadiusValue: string = "12px";
+        const rendered: any = mount(
+            <CSSBorderRadius
+                managedClasses={managedClasses}
+                onChange={jest.fn()}
+            />
+        );
+
+        rendered.setProps({
+            data: {
+                borderRadius: borderRadiusValue
+            },
+        });
+        rendered.setState({ indvidualValues: true });
```

Comment:
rather than setting state on this, it's preferable to test what action would cause this, so can you use a simulated onClick to the button that toggles this value